### PR TITLE
Parameterize default storageclass for ibm cloud

### PR DESCRIPTION
**Pull request by Tomcli**
For IBM Cloud IKS service, different machines such as classic and VCP have different default storageclass. Also, different Kubernetes version also have different storageclass due to the [recent IKS changes](https://cloud.ibm.com/docs/containers?topic=containers-kube_concepts#default_storageclass). Therefore, we want to parameterize the bash command to create a better user experience

### CODE REVIEW

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 1]
```diff
@@ -75,21 +75,23 @@ get the best experience from Kubeflow.
 
 6. Set the Block Storage as the default storageclass.
     ```shell
-    kubectl patch storageclass ibmc-block-gold -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+    NEW_STORAGE_CLASS=ibmc-block-gold
+    OLD_STORAGE_CLASS=$(kubectl get sc | grep "(default)" | grep -v ${NEW_STORAGE_CLASS} | awk '{ print $1;exit }')
```
Comment (shawnzhu) [inline content/en/docs/ibm/deploy/install-kubeflow.md:79]: Do you think this command is better? `kubectl get sc -o jsonpath='{.items[?(@.metadata.annotations.storageclass\\.kubernetes\\.io\/is-default-class=="true")].metadata.name}'` works on all OSes.

Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md:79]: The jsonpath command doesn't work on my machine, let me see is there a better way to grep it.

Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: @shawnzhu I updated the command to use jsonpath

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 2]
```diff
@@ -76,7 +76,7 @@ get the best experience from Kubeflow.
 6. Set the Block Storage as the default storageclass.
     ```shell
     NEW_STORAGE_CLASS=ibmc-block-gold
-    OLD_STORAGE_CLASS=$(kubectl get sc | grep "(default)" | grep -v ${NEW_STORAGE_CLASS} | awk '{ print $1;exit }')
+    OLD_STORAGE_CLASS=$(kubectl get sc -o jsonpath='{.items[?(@.metadata.annotations.storageclass\.kubernetes\.io\/is-default-class=="true")].metadata.name}')
```
Comment (shawnzhu) [inline content/en/docs/ibm/deploy/install-kubeflow.md:79]: 👍

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 3]
```diff
@@ -75,21 +75,23 @@ get the best experience from Kubeflow.
 
 6. Set the Block Storage as the default storageclass.
     ```shell
-    kubectl patch storageclass ibmc-block-gold -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
+    NEW_STORAGE_CLASS=ibmc-block-gold
+    OLD_STORAGE_CLASS=$(kubectl get sc -o jsonpath='{.items[?(@.metadata.annotations.storageclass\.kubernetes\.io\/is-default-class=="true")].metadata.name}')
+    kubectl patch storageclass ${NEW_STORAGE_CLASS} -p '{"metadata": {"annotations":{"storageclass.kubernetes.io/is-default-class":"true"}}}'
 
     # Check the default storageclass is block storage
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:82]: Change to `# Check if the default...` by adding "if"

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 4]
```diff
@@ -75,21 +75,23 @@ get the best experience from Kubeflow.
...
+    kubectl get storageclass | grep "(default)"
     ```
 
     Example output:
     ```
     ibmc-block-gold (default)   ibm.io/ibmc-block   65s
     ```
 
-    Make sure `ibmc-block-gold` is the only `default` storageclass. If there are two or more rows in the above output, there is other `default` storageclass. Unset it with the below command, for example, will make the `ibmc-file-bronze` storage no longer the `default` storageclass for the cluster.
+    Make sure `ibmc-block-gold` is the only `default` storageclass. If there are two or more rows in the above output, there is other `default` storageclass. Unset it with the below command, for example, to remove the previous `default` storageclass for the cluster.
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:91]: Let's use "`storageclass`" (one word surrounded by back ticks "`") or "storage class" (two words, like some of the hard-coded variables you use elsewhere)

Comment (8bitmp3) [reply content/en/docs/ibm/deploy/install-kubeflow.md:91]: Maybe we can change "...there is other `default` storageclass" to something like "...IBM Cloud assigns a different `default` storage class." to make the sentence sound better. WDYT?

Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:91]: Suggesting to replace "Unset it with ... to remove the previous `default` storageclass..." with "You can use the command below to remove the `default` storage class for the cluster:" (this eliminating some extra words - "unset... to remove" -> "remove")

WDYT?

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hi! I've made a few improvement suggestions through minor refactoring and enhancing the sentence syntax/grammar here. For example:

- Let's use "`storageclass`" (one word surrounded by back ticks "`") or "storage class" (two words, like some of the hard-coded variables you use elsewhere)

- Maybe we can change "...there is other `default` storageclass" to something like "...IBM Cloud assigns a different `default` storage class."

- Suggesting to replace "Unset it with ... to remove the previous `default` storageclass..." with "You can use the command below to remove the `default` storage class for the cluster:" (this eliminating some extra words - "unset... to remove" -> "remove")

File: content/en/docs/ibm/deploy/install-kubeflow.md (hunk 4 above)
Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: Thanks, I think previously the sentence was too wordy, so I removed the second half of the sentence to make it more concise.

File: content/en/docs/ibm/deploy/install-kubeflow.md (hunk 3 above)
Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: I changed it to `# List all the (default) storage classes`

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 5]
```diff
@@ -73,13 +73,13 @@ get the best experience from Kubeflow.
     kubectl get storageclasses | grep block
     ```
 
-6. Set the Block Storage as the default storageclass.
+6. Set the Block Storage as the default storageclass.
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:76]: Line 76 - same as in line 82: "storage class" (two words, instead of one)

`    # List all the (default) storage classes`

I also found that IBM uses two words in its docs: https://www.ibm.com/cloud/architecture/content/course/ibm-cloud-private-persistent-storage/defining-storage-classes:

> Default storage class
>
> A default storage class can dynamically provision storage when a storage class isn't specified.
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 6]
```diff
@@ -88,8 +88,7 @@ get the best experience from Kubeflow.
     ibmc-block-gold (default)   ibm.io/ibmc-block   65s
     ```
 
-    Make sure `ibmc-block-gold` is the only `default` storageclass. If there are two or more rows in the above output, there is other `default` storageclass. Unset it with the below command, for example, to remove the previous `default` storageclass for the cluster.
-
+    Make sure `ibmc-block-gold` is the only `(default)` storage class. If there are two or more rows in the above output, unset the previous `(default)` storage classes with the below command:
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:91]: Is there a reason why you have round brackets now around the word `default`?

Comment (8bitmp3) [reply content/en/docs/ibm/deploy/install-kubeflow.md:91]: `with the below command:` -> `with the command below:`

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hi @Tomcli A few things from my side:

- Line 76 - same as in line 82: let's use "storage class" (two words, instead of one)

`    # List all the (default) storage classes`

I also found that IBM uses two words in its docs: https://www.ibm.com/cloud/architecture/content/course/ibm-cloud-private-persistent-storage/defining-storage-classes:

> Default storage class
>
> A default storage class can dynamically provision storage when a storage class isn't specified.

- Lines 82, 91: Is there a reason why you have round brackets now around the word `default`? (I'm not familiar with IBM Cloud)

- Line 91: `with the below command:` -> `with the command below:`

Cheers!

File: content/en/docs/ibm/deploy/install-kubeflow.md (hunk 6 above)
Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: `ibmc-block-gold (default)` is how kubectl indicate the default storage class. However, some Kubernetes cluster also comes with a storage class called `default` (without parentheses) but is not the default storage class. Therefore I'm adding the parentheses here to avoid confusion.

File: content/en/docs/ibm/deploy/install-kubeflow.md (hunk 5 above)
Comment (Tomcli) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: thanks @8bitmp3

File: content/en/docs/ibm/deploy/install-kubeflow.md (hunk 6 above)
Comment (shawnzhu) [reply content/en/docs/ibm/deploy/install-kubeflow.md]: ☝️ is exactly what happens from my k8s cluster in IKS

**Review by animeshsingh — APPROVED** [review]
/lgtm

**Comment by Tomcli** [comment]
@shawnzhu is this PR look good to you?

**Comment by Tomcli** [comment]
@animeshsingh can you approve this PR? Thanks.

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Hi @Tomcli I have a few more suggestions for the Kubeflow deployment on IBM Cloud guide:

- Line 3: add "on IBM Cloud" to the description to make it explicit

```diff
- description = "Instructions for deploying Kubeflow with the shell"		
+ description = "Instructions for deploying Kubeflow on IBM Cloud"
```

("with the shell" may be implied, since you are not using the UI here?)

- Line 11: add a few words about the [IBM Cloud CLI](https://www.ibm.com/cloud/cli) through which you'll be logging in and configuring KF on IBM Cloud

```diff
- * Authenticating with IBM Cloud
+ * Authenticate with IBM Cloud
+
+   Log into IBM Cloud using the [IBM Cloud Command Line Interface (CLI)](https://www.ibm.com/cloud/cli) as follows:
...
```

- Lines 17-21: Refactor (also, IKS has been pre-defined)

```diff
- * Accessing the IBM Cloud cluster
+ * Create a Kubernetes cluster on IKS

-  If you do not have access to a cluster created with IBM Cloud Kubernetes Service, follow the [Create an IBM Cloud cluster](/docs/ibm/create-cluster) guide to create a cluster.
+   To deploy Kubeflow on IBM Cloud, you need a cluster running on IKS. If you don't have a cluster running, follow the [Create an IBM Cloud cluster](/docs/ibm/create-cluster) guide.

-   Run following command to switch the Kubernetes context and access the cluster.
+   Run the following command to switch the Kubernetes context and access the cluster:
...
```

- Line 38: change first person to second ("we" -> "you")

```diff
- Therefore, we strongly recommend to set up [IBM Cloud Block Storage](https://cloud.ibm.com/docs/containers?topic=containers-block_storage#add_block) as the default storage class so that you can
+ Therefore, you're recommended to set up [IBM Cloud Block Storage](https://cloud.ibm.com/docs/containers?topic=containers-block_storage#add_block) as the default storage class so that you can
...
```

**Comment by Tomcli** [comment]
Thanks @8bitmp3, I addressed your comments.

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 7]
```diff
@@ -1,25 +1,27 @@
...
+  Log into IBM Cloud using the [IBM Cloud Command Line Interface (CLI)](https://www.ibm.com/cloud/cli) as follows:
 
   ```shell
   ibmcloud login
   ```
 
-* Accessing the IBM Cloud cluster
-
-  If you do not have access to a cluster created with IBM Cloud Kubernetes Service, follow the [Create an IBM Cloud cluster](/docs/ibm/create-cluster) guide to create a cluster.
+* Create and access to a Kubernetes cluster on IKS
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:19]: Suggestion:

- Remove "to" in "access to":

```suggestion
* Create and access a Kubernetes cluster on IKS
```

(original suggestion:
```
+ * Create a Kubernetes cluster on IKS
```

File: content/en/docs/ibm/deploy/install-kubeflow.md [hunk 8]
```diff
@@ -48,18 +50,18 @@ get the best experience from Kubeflow.
 
 3. Install the IBM Cloud Block Storage plug-in. When you install the plug-in, pre-defined block storage classes are added to your cluster.
```
Comment (8bitmp3) [inline content/en/docs/ibm/deploy/install-kubeflow.md:51]: Suggestion:

- A blank line separator before a code block here and everywhere else.

**Review by 8bitmp3 — CHANGES_REQUESTED** [review]
Thank you @Tomcli . Just a few things (see comments).

**Comment by Tomcli** [comment]
Thanks @8bitmp3 for your reviews

**Comment by 8bitmp3** [comment]
@animeshsingh Can you please check the changes again and, if they're OK, approve the PR? 👏

**Comment by 8bitmp3** [comment]
@animeshsingh Thank you so much!
